src/execution/compiler/pipeline.cpp

```diff
@@ -238,28 +238,28 @@ ast::FunctionDecl *Pipeline::GenerateRunPipelineFunction(query_id_t query_id) co
       op->BeginPipelineWork(*this, &builder);
     }
 
+    // var pipelineState = @tlsGetCurrentThreadState(...)
+    auto exec_ctx = compilation_context_->GetExecutionContextPtrFromQueryState();
+    auto tls = codegen_->ExecCtxGetTLS(exec_ctx);
+    auto state = codegen_->TLSAccessCurrentThreadState(tls, state_.GetTypeName());
```

Comment:
Why was this moved up?

Reply:
Actually looking at this again, I think it's a mistake. I needed the `pipelineState` inside of `runPipeline` even in the parallel case, because I needed to pass it to leftOuterConsumer. But I think this is rare. I'll revert the changes and move this code into hash_join_translator.cpp.